### DISCUSSION TITLE
Compatibility with KitQL

### DISCUSSION THREAD

**Discussion by PaulVlas** [vendurehq/vendure#1667] [General]
Hi,

I've been looking at creating a frontend for Venture using Svelte Kit. I've been trying out several libraries for communicating with the GraphQL API from SSR. I also saw an example of [Sveltekit with Vendure](https://github.com/spences10/sveltekit-vendure-commerce).

The problem is that this version of KitQL is deprecated. The new version uses Houdini to communicate with GraphQL. However, when I try to generate the houdini client, it gives me this error message:

> Your project defines a Node interface but it does not conform to the Global Identification Spec.
> 
> If you are trying to provide the Node interface and its field, they must look like the following:
> interface Node {
>         id: ID!
> }
> 
> extend type Query {
>         node(id: ID!): Node
> }
> 
> For more information, please visit these links:
> - https://graphql.org/learn/global-object-identification/
> - https://www.houdinigraphql.com/guides/caching-data#custom-ids

I have not yet tried fiddling around and manually adding a node thing in the graphql query. I don't know if this is easy to add or not, but it would be appreciated if Vendure would be made compatible with this version of KitQL so it is easy to use with SvelteKit. (it is one of the few libraries supporting SSR with GraphQL out of the box).

Thank you for taking the time to create this wonderful application and I'm looking forward to see more features being added!

**Comment by michaelbromley** [comment]
Hi!
Good timing, we were just discussing this earlier today. Please see this thread: https://github.com/spences10/sveltekit-vendure-commerce/pull/129#issuecomment-1189933815

So I think hopefully that PR will get merged soon and you'll be able to work around these issues.

**Reply by PaulVlas** [reply]
Owh, what a coincidence. I managed to work around the issue by [changing the way houdini resolves the types](https://www.houdinigraphql.com/guides/caching-data#custom-ids).
